src/views/FlexExampleAppHomeView.mxml

```diff
@@ -1,204 +1,129 @@
-				trace("[INCENTIVIZED] FETCH_FAILED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_COMPLETED, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_COMPLETED");
-				return null;
-			});
-			
-			IncentivizedAd.getInstance().addEventListener(HeyzapEvent.IS_INCOMPLETE, function(event:HeyzapEvent=null) {
-				trace("[INCENTIVIZED] IS_INCOMPLETE");
-				return null;
-			});
-			
-		]]>
+        import com.heyzap.sdk.ads.BannerAd;
+        import com.heyzap.sdk.ads.HeyzapAds;
+        import com.heyzap.sdk.ads.IncentivizedAd;
+        import com.heyzap.sdk.ads.InterstitialAd;
+        import com.heyzap.sdk.ads.VideoAd;
+
+        import model.AppModel;
+
+        [Bindable]
+        public var appModel:AppModel;
+
+        private var currentActionObject:*;
+
+        override public function set data(value:Object):void {
+            appModel = value as AppModel;
+        }
+
+        private function onCreationComplete():void {
+            currentActionObject = InterstitialAd.getInstance();
+        }
+
+        private function getEnteredAdTagOrNull():String{
+            if (!adTagTextInput.text) return null;
+            else return adTagTextInput.text;
+        }
+
+        private function onFetchButtonClick():void {
+            currentActionObject.fetch(getEnteredAdTagOrNull());
+        }
+
+        private function onHideButtonClick():void {
+            currentActionObject.hide();
+            appModel.bannerIsShown = false;
+        }
+
+        private function onShowButtonClick():void {
+            if (currentActionObject instanceof  BannerAd) {
+                currentActionObject.show(BannerAd.POSITION_BOTTOM,getEnteredAdTagOrNull()); //todo add position
+                appModel.bannerIsShown = true;
+            } else {
+                currentActionObject.show(getEnteredAdTagOrNull());
+            }
+        }
+
+        private function onAvailableButtonClick():void {
+            logConsole.showLine("Available? result: " + currentActionObject.isAvailable(getEnteredAdTagOrNull()));
+        }
+
+        private function enableAllButtons():void {
+            fetchButton.enabled = true;
+            showButton.enabled = true;
+            availableButton.enabled = true;
+        }
+
+        private function onActionTypeGroupSelectionChange(event:Event):void {
+            enableAllButtons();
+
+            if (interstitialRadioButton.selected) {
+                currentActionObject = InterstitialAd.getInstance();
+            } else if (incentivizedRadioButton.selected) {
+                currentActionObject = IncentivizedAd.getInstance();
+            } else if (videoRadioButton.selected) {
+                currentActionObject = VideoAd.getInstance();
+            } else if (bannerRadioButton.selected) {
+                currentActionObject = BannerAd.getInstance();
```

Comment:
again, state may be an issue here, because it is valid to show a video or interstitial while a banner is also showing. using the same button for showing/hiding a banner as showing another ad is problematic for this reason.